src/main/java/edu/illinois/library/cantaloupe/async/ThreadPool.java

```diff
@@ -66,11 +64,11 @@ String getThreadNamePrefix() {
 
     private boolean isShutdown = false;
     private final ExecutorService lowPriorityPool =
-            Executors.newCachedThreadPool(new LowPriorityThreadFactory());
+            Executors.newThreadPerTaskExecutor(new LowPriorityThreadFactory());
     private final ExecutorService normalPriorityPool =
-            Executors.newCachedThreadPool(new NormalPriorityThreadFactory());
+            Executors.newThreadPerTaskExecutor(new NormalPriorityThreadFactory());
     private final ExecutorService highPriorityPool =
-            Executors.newCachedThreadPool(new HighPriorityThreadFactory());
+            Executors.newThreadPerTaskExecutor(new HighPriorityThreadFactory());
```

Comment:
Hits [these JMH benchmarks](https://github.com/ponder-lab/cantaloupe/blob/d34f2594b1257dd9795d61863bc8aa55340477a9/src/test/java/edu/illinois/library/cantaloupe/perf/processor/FfmpegProcessorPerformance.java#L53-L111); local results are very similar for legacy and virtual